game-core/src/main/java/games/strategy/triplea/attachments/TechAbilityAttachment.java

```diff
@@ -102,25 +109,72 @@ public TechAbilityAttachment(final String name, final Attachable attachable, fin
     super(name, attachable, gameData);
   }
 
-  // setters and getters
+  private UnitType getUnitType(final String value) throws GameParseException {
+    return Optional.ofNullable(getData().getUnitTypeList().getUnitType(value))
+        .orElseThrow(() -> new GameParseException("No unit called:" + value + thisErrorMsg()));
+  }
+
+  private String[] splitAndValidate(final String name, final String value) throws GameParseException {
```

Comment:
Some unit tests for these new methods would be useful to ensure they operate correctly with various inputs.  That might require creating new static overloads so you can test them without depending on instance methods like `thisErrorMsg()`.

Reply:
Mockito has a solution for that ^^